Description: Create disk/by-partlabel links for mmcblk partitions
 Android based kernel exports the uevent property PARTNAME, which can be
 used to find out at run time the named partitions (e.g. boot) for the
 device.
 .
 This is specially useful for the Touch based images and flash-kernel,
 to automatically update the kernel by writing at the correct partition
 (independently of the hardware revision).
Forwarded: no
Author: Ricardo Salveti <ricardo.salveti@canonical.com>

Index: systemd/rules/60-persistent-storage.rules
===================================================================
--- systemd.orig/rules/60-persistent-storage.rules	2013-01-30 04:32:08.399835311 +0100
+++ systemd/rules/60-persistent-storage.rules	2013-05-22 11:07:07.486872081 +0200
@@ -48,6 +48,7 @@
 
 KERNEL=="mmcblk[0-9]", SUBSYSTEMS=="mmc", ATTRS{name}=="?*", ATTRS{serial}=="?*", ENV{ID_NAME}="$attr{name}", ENV{ID_SERIAL}="$attr{serial}", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}"
 KERNEL=="mmcblk[0-9]p[0-9]", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}-part%n"
+KERNEL=="mmcblk[0-9]p[0-9]", ENV{PARTNAME}=="?*", SYMLINK+="disk/by-partlabel/$env{PARTNAME}"
 KERNEL=="mspblk[0-9]", SUBSYSTEMS=="memstick", ATTRS{name}=="?*", ATTRS{serial}=="?*", ENV{ID_NAME}="$attr{name}", ENV{ID_SERIAL}="$attr{serial}", SYMLINK+="disk/by-id/memstick-$env{ID_NAME}_$env{ID_SERIAL}"
 KERNEL=="mspblk[0-9]p[0-9]", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/memstick-$env{ID_NAME}_$env{ID_SERIAL}-part%n"
 
